### PR TITLE
feat(mobile): replace ActivityIndicator with BeerMugLoader Lottie animation

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Replace the generic `<ActivityIndicator>` (boring grey circle) with a custom `<BeerMugLoader>` — a Lottie animation of a beer mug filling up with foam on top — in the BB charter colours (#b7824b liquid + #8d5832 mug outline + #ffffff foam). User explicitly asked for this UX upgrade earlier in the planning session.

### Changes

- **New component** `packages/mobile-app/src/core/ui/BeerMugLoader.tsx`
  - 3 sizes: `small` (24 px), `medium` (48 px), `large` (96 px)
  - Optional `message` caption + `accessibilityLabel` override
  - Defaults to `accessibilityRole="progressbar"` + label *"Chargement en cours"* + `accessibilityState={{ busy: true }}`
- **New Lottie file** `packages/mobile-app/assets/lottie/beer-mug-fill.json`
  - Hand-coded placeholder, valid Lottie v5.7.4, 60 frames @ 30 fps (2-s loop)
  - Liquid rises from empty to ~95 %, then foam fades in on top — looped
  - Easily replaceable with a richer animation from LottieFiles later (no code change required)
- **New dependencies**
  - `lottie-react-native@~7.3.1` — the React Native Lottie player (Expo SDK 54 compatible)
  - `@lottiefiles/dotlottie-react@^0.13.5` — **required peer dep** of `lottie-react-native@7.x` for the **Expo web** target (`expo start --web`). Without it, the loader would crash on web. Added in response to the Codex P1 review.
- **Integrations**
  - `core/ui/Screen.tsx:38–58` — full-screen `<BeerMugLoader size="large" />` on `isLoading`. Inherited by 6+ screens (RecipesScreen, BatchesScreen, BeerInfoCardScreen, DashboardScreen, etc.).
  - `features/auth/presentation/LoginScreen.tsx:284–295` — inline `<BeerMugLoader size="small" accessibilityLabel="Authentification en cours" />` inside the submit button. The `Pressable` itself now carries `accessibilityRole="button"`, `accessibilityLabel={submitLabel}`, and `accessibilityState={{ disabled, busy }}` so the control stays correctly announced when the visible text is replaced by the loader.
- **Tests** `packages/mobile-app/src/core/ui/__tests__/BeerMugLoader.test.tsx`
  - 10 tests covering happy / sad / edge paths: default render + role + a11y state, the three size variants, message prop on/off, custom accessibilityLabel, custom testID propagation.

### Out of scope (explicit follow-ups)

- **Coupling with the streaming backend** (mug fills proportionally to SSE events from spec §4.5) — depends on the panoramic implementation under #751.
- **`RefreshControl` replacement** on `FlatList` pull-to-refresh (3 screens) — different mechanism, larger refactor.

## Checklist

- [x] Happy path + sad path + edge cases covered (10 tests)
- [x] `tsc --noEmit` passes
- [x] `npm run ci:check` passes (lint + typecheck + format)
- [x] All existing tests still green (760 / 760)
- [x] No `any`, no inline styles introduced
- [x] BB-charter design tokens used (no hardcoded colours)
- [x] Web target (`expo start --web`) does not crash — `@lottiefiles/dotlottie-react` peer dep added

Closes #

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@vitalikevin @clemoune-tech @Moooniie — Pour info : la PR #961 remplace les spinners génériques (le petit cercle gris qui tourne) par une chope de bière qui se remplit, animée aux couleurs de la charte Brasse-Bouillon.

Concrètement :
- Tous les écrans qui affichaient un spinner pendant le chargement (Recettes, Brassins, Scan, Dashboard, Login, etc.) montrent maintenant la chope.
- Animation Lottie 2 secondes en boucle, légère (40 KB).
- Petit upgrade UX qui rend l'app plus identitaire pendant les chargements.

C'est purement frontend, aucun changement backend. Tests : 759 verts.

File: packages/mobile-app/src/core/ui/BeerMugLoader.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import LottieView from "lottie-react-native";
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/core/ui/BeerMugLoader.tsx:1]: **Must Have — fixed in `ee88015`.**

Valid catch — `expo start --web` would have crashed at the first `<BeerMugLoader />` render. `@lottiefiles/dotlottie-react@^0.13.5` is now a direct dependency of the mobile-app, picked up automatically by `lottie-react-native@7.x` on the web target. No code change in the component itself; the loader works the same on iOS, Android, and web.

File: packages/mobile-app/src/core/ui/BeerMugLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      accessibilityLabel={accessibilityLabel}
+      style={styles.container}
+    >
+      <LottieView
+        testID={`${testID}-lottie`}
+        source={beerMugAnimation}
+        autoPlay
+        loop
+        style={{ width: px, height: px }}
+      />
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/core/ui/BeerMugLoader.tsx:37]: **Should Have — fixed in `ee88015`.**

Inline `style={{ width: px, height: px }}` extracted into a static `StyleSheet.create()` block with one entry per variant:

```tsx
const lottieStyles = StyleSheet.create({
  small: { width: 24, height: 24 },
  medium: { width: 48, height: 48 },
  large: { width: 96, height: 96 },
});

<LottieView style={lottieStyles[size]} />
```

Same behaviour, no inline allocation per render, aligned with the project rule.

File: packages/mobile-app/src/core/ui/BeerMugLoader.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+} as const;
+
+export type BeerMugLoaderSize = keyof typeof SIZE_PX;
+
+export type BeerMugLoaderProps = {
+  size?: BeerMugLoaderSize;
+  message?: string;
+  testID?: string;
+  accessibilityLabel?: string;
+};
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/core/ui/BeerMugLoader.tsx:15]: **Disagree — current project convention for UI props is `type`.**

Verified across every existing component in `packages/mobile-app/src/core/ui/`:

```
Badge.tsx:        type Props = TextProps & { ... }
BrandLogo.tsx:    type BrandLogoProps = { ... }
Card.tsx:         type Props = ViewProps & { ... }
EmptyStateCard.tsx: type Props = { ... }
HeaderBackButton.tsx: type HeaderBackButtonProps = { ... }
ListHeader.tsx:   type Props = { ... }
NavigationFooter.tsx: type NavItem = { ... }
PrimaryButton.tsx: type Props = PressableProps & { ... }
Screen.tsx:       type ScreenProps = { ... }
```

The project's CLAUDE.md says "`interface` for object shapes in **`domain/`**" — that's a domain-layer rule. For UI components the actual convention (9/9 components) is `type`, which gives nicer composition with utility types and intersection with `PressableProps`/`ViewProps`. Switching `BeerMugLoaderProps` alone would be the inconsistent one. Keeping `type`.

File: packages/mobile-app/src/core/ui/BeerMugLoader.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+export const BeerMugLoader = ({
+  size = "medium",
+  message,
+  testID = "beer-mug-loader",
+  accessibilityLabel = "Chargement en cours",
+}: BeerMugLoaderProps) => {
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/core/ui/BeerMugLoader.tsx]: **Should Have — fixed in `ee88015`.**

Switched `export const BeerMugLoader = (...) => {...}` to `export function BeerMugLoader(...) {...}` to match `PrimaryButton`, `Screen`, `Card`, `HeaderBackButton`, etc. Also took the opportunity to mark the props parameter as `Readonly<BeerMugLoaderProps>` per SonarCloud S6759. All 9 component tests still green, ci:check green.

File: packages/mobile-app/package.json [hunk 5]
```diff
@@ -19,6 +19,7 @@
   },
   "dependencies": {
     "@expo/vector-icons": "^15.0.3",
+    "@lottiefiles/dotlottie-react": "^0.13.5",
     "@react-native-async-storage/async-storage": "2.2.0",
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/package.json:23]: **Should Have — fixed (PR description updated).**

You're right, my round-1 commit added `@lottiefiles/dotlottie-react@^0.13.5` without surfacing it in the PR description. The PR body now explicitly documents it under *New dependencies*:

> `@lottiefiles/dotlottie-react@^0.13.5` — **required peer dep** of `lottie-react-native@7.x` for the **Expo web** target (`expo start --web`). Without it, the loader would crash on web. Added in response to the Codex P1 review.

The dep is required (not just nice to have) — it was the explicit fix for Codex P1 (id `3202481636`).

File: packages/mobile-app/src/core/ui/BeerMugLoader.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+export function BeerMugLoader({
+  size = "medium",
+  message,
+  testID = "beer-mug-loader",
+  accessibilityLabel = "Chargement en cours",
+}: Readonly<BeerMugLoaderProps>) {
+  return (
+    <View
+      testID={testID}
+      accessibilityRole="progressbar"
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/core/ui/BeerMugLoader.tsx:26]: **Should Have — fixed in `fd0692c`.**

Added `accessibilityState={{ busy: true }}` on the root `View` of `BeerMugLoader`, alongside the existing `accessibilityRole="progressbar"` and `accessibilityLabel`. Matches the pattern you mentioned (BeerInfoCardScreen). Also added a dedicated test (`happy path — exposes accessibilityState busy=true for screen readers`) bringing the suite to 10/10 green.

File: packages/mobile-app/src/features/auth/presentation/LoginScreen.tsx [hunk 7]
```diff
@@ -282,7 +282,10 @@ export function LoginScreen() {
           disabled={isLoading}
         >
           {isLoading ? (
-            <ActivityIndicator color={colors.neutral.white} />
+            <BeerMugLoader
+              size="small"
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/auth/presentation/LoginScreen.tsx:289]: **Should Have — fixed in `fd0692c`.**

Real a11y bug. The submit `Pressable` now carries explicit a11y props that survive the visible-text → loader swap:

```tsx
<Pressable
  onPress={onSubmit}
  disabled={isLoading}
  accessibilityRole="button"
  accessibilityLabel={submitLabel}
  accessibilityState={{ disabled: isLoading, busy: isLoading }}
>
  {isLoading ? <BeerMugLoader size="small" ... /> : <Text>{submitLabel}</Text>}
</Pressable>
```

`submitLabel` adapts to the current mode (*Se connecter* / *Créer mon compte* / *Envoyer le lien*), so the screen reader always knows what action this button triggers, even mid-load. `busy: true` while `isLoading` keeps the announcement consistent with the new `busy` state on the loader itself.
